Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -66,15 +66,28 @@ export class azureclitask {
 
             tool.line(args); // additional args should always call line. line() parses quoted arg strings
 
+            var resourceGroupName: string = tl.getEndpointDataParameter(connectedService, "ResourceGroupName", true);
+            var mlWorkspaceName: string = tl.getEndpointDataParameter(connectedService, "MLWorkspaceName", true);
+
             var addSpnToEnvironment = tl.getBoolInput("addSpnToEnvironment", false);
             if (!!addSpnToEnvironment && tl.getEndpointAuthorizationScheme(connectedService, true) == "ServicePrincipal") {
                 await tool.exec({
                     failOnStdErr: failOnStdErr,
-                    env: { ...process.env, ...{ servicePrincipalId: this.servicePrincipalId, servicePrincipalKey: this.servicePrincipalKey } }
+                    env: {
+                        ...process.env, ...{
+                            servicePrincipalId: this.servicePrincipalId,
+                            servicePrincipalKey: this.servicePrincipalKey,
+                            resourceGroupName: resourceGroupName,
+                            mlWorkspaceName: mlWorkspaceName
+                        }
```

Comment:
This doesn't seem scalable. Instead of this, you could add the whole stringified endpoint object to the environment variable. And optionally let the user opt-in?

Reply:
keeping this scoped to ML data for now